### ISSUE TITLE
Problem with model training

### ISSUE THREAD

**Issue by nightingale233** [mit-han-lab/litepose#40]
I think your paper's mind is pretty good and wanted to have a train myself.
I downloaded litepose source code and use "Normal Training" method to trained a mobileNet based Search-S model on Crowdpose dataset for 500 epoches.
However,I only got 46.6 AP instead of 58.3 AP on CrowdPose Evaluation(as Fig 1 shows). 
<img width="582" alt="2ef1965e83ffd94d052757f62d54793" src="https://github.com/user-attachments/assets/72bb3cdd-1e3b-4a35-be41-d694333626b5">
I have changed INPUT_SIZE from 256 to 448 and OUTPUT_SIZE from [64,128] to [112,224] as your NOTE said.And to make training faster I changed IMAGES_PER_GPU from 16 to 32, GPU_EORKERS from 4 to 16. And that is all.
I also trained a Search-XS  model base on mobileNet and got 46.4 AP instead of 49.3 AP on CrowdPose Evaluation(as Fig 2 shows).
<img width="551" alt="89436a44d0d4cc6a4ea82cfae7ec88b" src="https://github.com/user-attachments/assets/1070683b-a429-4ecf-85a2-18a25f6d1fad">
I tried to output both official LitePose-Auto-S and my trained model in onnx format and used Netron to see the difference between each other but found they are totally the same.
My training env is on CUDA 11.8, torch 2.0.0 on AutoDL Server(1 RTX3090 24GB GPU).
Would someone help me fix this problem? Thank you very much!!!

**Comment by nightingale233** [comment]
My mistake. I will do supernet train first and then have a test.

**Comment by suiyingliuxin** [comment]
> My mistake. I will do supernet train first and then have a test.我的错误。我将先进行超网训练，然后进行测试。
Could you elaborate on exactly how to do it? I'm encountering the same problem.

**Comment by suiyingliuxin** [comment]
> My mistake. I will do supernet train first and then have a test.

Could you elaborate on exactly how to do it? I'm encountering the same problem.

**Comment by nightingale233** [comment]
Hello! I think you may understand CHINESE so I will return to you in CHINESE.
在github上面有如图所示的一句话，需要在进行超网训练之间先导入预训练模型，这个对应论文当中5.2的Pre-train Details.在复现实验的时候可以直接用官方的这个权重进行超网训练。具体是修改supermobile.yaml文件当中的
&nbsp; PRETRAINED: './pretrain/search_pretrain.pth.tar'

替换为：你下载的这个权重文件。然后按照论文要求进行800或者2400轮的训练。
python dist_train.py --cfg experiments/crowd_pose/mobilenet/supermobile.yaml
然后把权重按对应的结构进行转移。
python weight_transfer.py --cfg experiments/crowd_pose/mobilenet/supermobile.yaml --superconfig mobile_configs/search-XS.json TEST.MODEL_FILE your_supernet_checkpoint_path
随后再把训练完的权重，填写到mobile.yaml对应的位置里，再进行后续的调优训练。
python dist_train.py --cfg experiments/crowd_pose/mobilenet/mobile.yaml --superconfig mobile_configs/search-XS.json
最后对训练完的结果进行测试。
————————————————————————————————————————————————————————————
下面是我自己实验测试的结果：
使用这种方式，确实会取得比我直接训练好的结果，但是，依然没有达到论文的标称值。我询问了先前做实验并提问的人，他告诉我可以增加调优轮数，结果会更高一些。但似乎最终也没有达到论文的标称值。XS结构的最终AP大概在48.5左右。
————————————————————————————————————————————————————————————
如果你想要对这个模型做改进的话，你可以在网络结构上加上你自己改进的部分，然后带上官方结果权重进去训练。我看到有一篇论文LiteDEKR就是这么实验的。
LiteDEKR: End‐to‐end lite 2D human pose estimation network
————————————————————————————————————————————————————————————
希望能帮助到你！




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2025年4月5日(星期六) 晚上8:38
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [mit-han-lab/litepose] Problem with model training (Issue #40)












    
My mistake. I will do supernet train first and then have a test.我的错误。我将先进行超网训练，然后进行测试。
 Could you elaborate on exactly how to do it? I'm encountering the same problem.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  suiyingliuxin left a comment (mit-han-lab/litepose#40)
  
My mistake. I will do supernet train first and then have a test.我的错误。我将先进行超网训练，然后进行测试。
 Could you elaborate on exactly how to do it? I'm encountering the same problem.

**Comment by suiyingliuxin** [comment]
非常非常感谢你的解答，解决了我最近两周一直困惑的问题,我还有一些其他的疑问，如果你能够帮忙解答，我会万分感谢，第一个问题是，我看到mobile_configs有prune-L和search-L，如果我先进行超网训练或者加载官网提供的权重，是不是都是用search-系列的配置文件，第二个问题是，如果我对网络结构进行了改进，加载官方的权重训练会不会因为结构不符合而报错，应该如何解决呢，难道需要重新进行超网训练嘛?




Thank you so much for your answer. It has solved the problem that has been bothering me for the past two weeks.I still have some other questions. If you could help me answer them, I would be extremely grateful. The first question is that I see there are "prune-L" and "search-L" in "mobile configs". If I conduct super network training first or load the weights provided on the official website, should I use the "search-" series configuration files? The second question is that if I make improvements to the network structure, will loading the official weights for training result in an error due to the structure mismatch? How should I solve this problem? Do I need to conduct super network training again?













在 2025-04-05 21:11:20，"LoveLiz4Ever" ***@***.***> 写道：

Hello! I think you may understand CHINESE so I will return to you in CHINESE. <br> 在github上面有如图所示的一句话，需要在进行超网训练之间先导入预训练模型，这个对应论文当中5.2的Pre-train Details.在复现实验的时候可以直接用官方的这个权重进行超网训练。具体是修改supermobile.yaml文件当中的 <br> &amp;nbsp; PRETRAINED: &#39;./pretrain/search_pretrain.pth.tar&#39; <br> <br> 替换为：你下载的这个权重文件。然后按照论文要求进行800或者2400轮的训练。 <br> python dist_train.py --cfg experiments/crowd_pose/mobilenet/supermobile.yaml <br> 然后把权重按对应的结构进行转移。 <br> python weight_transfer.py --cfg experiments/crowd_pose/mobilenet/supermobile.yaml --superconfig mobile_configs/search-XS.json TEST.MODEL_FILE your_supernet_checkpoint_path <br> 随后再把训练完的权重，填写到mobile.yaml对应的位置里，再进行后续的调优训练。 <br> python dist_train.py --cfg experiments/crowd_pose/mobilenet/mobile.yaml --superconfig mobile_configs/search-XS.json <br> 最后对训练完的结果进行测试。 <br> ———————————————————————————————————————————————————————————— <br> 下面是我自己实验测试的结果： <br> 使用这种方式，确实会取得比我直接训练好的结果，但是，依然没有达到论文的标称值。我询问了先前做实验并提问的人，他告诉我可以增加调优轮数，结果会更高一些。但似乎最终也没有达到论文的标称值。XS结构的最终AP大概在48.5左右。 <br> ———————————————————————————————————————————————————————————— <br> 如果你想要对这个模型做改进的话，你可以在网络结构上加上你自己改进的部分，然后带上官方结果权重进去训练。我看到有一篇论文LiteDEKR就是这么实验的。 <br> LiteDEKR: End‐to‐end lite 2D human pose estimation network <br> ———————————————————————————————————————————————————————————— <br> 希望能帮助到你！ <br> <br> <br> <br> <br> ------------------&amp;nbsp;原始邮件&amp;nbsp;------------------ <br> 发件人: ***@***.***&amp;gt;; <br> 发送时间: 2025年4月5日(星期六) 晚上8:38 <br> 收件人: ***@***.***&amp;gt;; <br> 抄送: ***@***.***&amp;gt;; ***@***.***&amp;gt;; <br> 主题: Re: [mit-han-lab/litepose] Problem with model training (Issue #40) <br> <br> <br> <br> <br> <br> <br> <br> <br> <br> <br> <br> <br> <br> My mistake. I will do supernet train first and then have a test.我的错误。我将先进行超网训练，然后进行测试。 <br> Could you elaborate on exactly how to do it? I&#39;m encountering the same problem. <br> <br> — <br> Reply to this email directly, view it on GitHub, or unsubscribe. <br> You are receiving this because you authored the thread.Message ID: ***@***.***&amp;gt; <br> suiyingliuxin left a comment (mit-han-lab/litepose#40) <br> <br> My mistake. I will do supernet train first and then have a test.我的错误。我将先进行超网训练，然后进行测试。 <br> Could you elaborate on exactly how to do it? I&#39;m encountering the same problem. <br> <br> — <br> Reply to this email directly, view it on GitHub, or unsubscribe. <br> You are receiving this because you authored the thread.Message ID: ***@***.***&amp;gt;

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: ***@***.***>

nightingale233 left a comment (mit-han-lab/litepose#40)
Hello! I think you may understand CHINESE so I will return to you in CHINESE. <br> 在github上面有如图所示的一句话，需要在进行超网训练之间先导入预训练模型，这个对应论文当中5.2的Pre-train Details.在复现实验的时候可以直接用官方的这个权重进行超网训练。具体是修改supermobile.yaml文件当中的 <br> &amp;nbsp; PRETRAINED: &#39;./pretrain/search_pretrain.pth.tar&#39; <br> <br> 替换为：你下载的这个权重文件。然后按照论文要求进行800或者2400轮的训练。 <br> python dist_train.py --cfg experiments/crowd_pose/mobilenet/supermobile.yaml <br> 然后把权重按对应的结构进行转移。 <br> python weight_transfer.py --cfg experiments/crowd_pose/mobilenet/supermobile.yaml --superconfig mobile_configs/search-XS.json TEST.MODEL_FILE your_supernet_checkpoint_path <br> 随后再把训练完的权重，填写到mobile.yaml对应的位置里，再进行后续的调优训练。 <br> python dist_train.py --cfg experiments/crowd_pose/mobilenet/mobile.yaml --superconfig mobile_configs/search-XS.json <br> 最后对训练完的结果进行测试。 <br> ———————————————————————————————————————————————————————————— <br> 下面是我自己实验测试的结果： <br> 使用这种方式，确实会取得比我直接训练好的结果，但是，依然没有达到论文的标称值。我询问了先前做实验并提问的人，他告诉我可以增加调优轮数，结果会更高一些。但似乎最终也没有达到论文的标称值。XS结构的最终AP大概在48.5左右。 <br> ———————————————————————————————————————————————————————————— <br> 如果你想要对这个模型做改进的话，你可以在网络结构上加上你自己改进的部分，然后带上官方结果权重进去训练。我看到有一篇论文LiteDEKR就是这么实验的。 <br> LiteDEKR: End‐to‐end lite 2D human pose estimation network <br> ———————————————————————————————————————————————————————————— <br> 希望能帮助到你！ <br> <br> <br> <br> <br> ------------------&amp;nbsp;原始邮件&amp;nbsp;------------------ <br> 发件人: ***@***.***&amp;gt;; <br> 发送时间: 2025年4月5日(星期六) 晚上8:38 <br> 收件人: ***@***.***&amp;gt;; <br> 抄送: ***@***.***&amp;gt;; ***@***.***&amp;gt;; <br> 主题: Re: [mit-han-lab/litepose] Problem with model training (Issue #40) <br> <br> <br> <br> <br> <br> <br> <br> <br> <br> <br> <br> <br> <br> My mistake. I will do supernet train first and then have a test.我的错误。我将先进行超网训练，然后进行测试。 <br> Could you elaborate on exactly how to do it? I&#39;m encountering the same problem. <br> <br> — <br> Reply to this email directly, view it on GitHub, or unsubscribe. <br> You are receiving this because you authored the thread.Message ID: ***@***.***&amp;gt; <br> suiyingliuxin left a comment (mit-han-lab/litepose#40) <br> <br> My mistake. I will do supernet train first and then have a test.我的错误。我将先进行超网训练，然后进行测试。 <br> Could you elaborate on exactly how to do it? I&#39;m encountering the same problem. <br> <br> — <br> Reply to this email directly, view it on GitHub, or unsubscribe. <br> You are receiving this because you authored the thread.Message ID: ***@***.***&amp;gt;

**Comment by suiyingliuxin** [comment]
I encountered some issues during my super network training. Could you please elaborate on your super network configuration in detail? Also, is the pre-trained model you loaded the crowdpose.path.tar provided on the official website? I didn't apply learning rate decay. I found that my super network AP reached 19 at 350 rounds, but then it kept declining to 16 at 700 rounds. How can I solve these problems?

**Comment by nightingale233** [comment]
Hello! I think you should read training part of this paper again. You can send e-mail to 3552993160@qq.com if there is any other problem.
